app/models/concerns/operations/stock_operations.rb

```diff
@@ -33,11 +33,10 @@ def inventorize(package, location)
     #
     # @raise [Goodcity::UninventoryError] thrown if actions were taken since the initial inventory action
     #
-    # @param [Package|String] the package to inventorize or its id
     # @param [Location|String] the location to place the package in
     #
-    def uninventorize(package)
-      last_action = PackagesInventory.order('id DESC').where(package: package).limit(1).first
+    def uninventorize(package_inventory_id)
+      last_action = PackagesInventory.find(package_inventory_id)
```

Comment:
@swatijadhav Could you explain this change ?

This looks wrong to me. A package can only be un-inventorized right after the inventory action. But that logic should remain encapsulated in the operations concern. This extracts it out and opens the doors for misuse

Reply:
Added this change initially, while working on `undo` actions. But as discussed we are not covering undo actions in this PR, I will revert this change.

Reply:
Thanks :)

Note: even in the case of an undo feature, this change would be wrong. We can discuss a proper undo/redo design when the time comes